bpf/bpf_host.c

```diff
@@ -428,10 +438,13 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx, bool from_host)
 	}
 
 #ifdef ENABLE_HOST_FIREWALL
-	if (from_host)
-		ret = ipv6_host_policy_egress(ctx, secctx);
-	else
+	if (from_host) {
+		/* Bypass host policies if on return path from proxy. */
+		if (!is_from_proxy(ctx))
```

Comment:
I'm surprised that this is needed. Shouldn't `secctx` be the actual security identity other than `HOST_ID` and thus lead to an allow?

Reply:
That is correct and the commit that added bypassing of the host policies for the L7 proxy actually isn't needed. See https://github.com/cilium/cilium/pull/11507#discussion_r426893006 for details. I've removed it.